test/parallel/test-assert-async.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const promisify = require('util').promisify;
+const wait = promisify(setTimeout);
+
+// Ensure async support for assert.rejects() and assert.doesNotReject()
```

Comment:
Hm, I would say it tests the functionality of `assert.rejects()` and `assert.doesNotReject()` but not that it "ensures" async support for those as they should always be async.

I would describe it as e.g. "Tests `assert.rejects()` and `assert.doesNotReject()` by checking their expected output and by verifying that they do not work sync."